Add information on killing MeiliSearch (#690)
meilisearch/documentation: guides/advanced_guides/asynchronous_updates.md

  "processedAt": "2019-12-07T20:23:50.157436246Z"
}
```

> ### Killing MeiliSearch When a Task is Processing

Reviewer: ```suggestion
### Killing MeiliSearch While a Task is Processing
```

> Since in MeiliSearch asynchronous task are [atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)), killing MeiliSearch in the middle of the process does not corrupt or alter the database.

Reviewer: ```suggestion
Since in MeiliSearch asynchronous tasks are [atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)), killing MeiliSearch in the middle of a process does not corrupt or alter the database.
```

> > An atomic transaction is an indivisible and irreducible series of database operations such that either all occur, or nothing occurs.

Reviewer: I think we should add this line to the glossary instead of linking to Wikipedia.

> Essentially, tasks are done in transactions. If the transaction for any reason fails or is killed in the middle of its process, nothing of the tasks will be committed to your database.

Reviewer: ```suggestion
Essentially, tasks are done in transactions. If the transaction fails or is killed for any reason before completing, none of the tasks will be committed to your database.
```

> Depending on the status of each process that you can know [by using the update route](/references/updates.md), you can determine if a process has been committed to MeiliSearch or not.

Reviewer: ```suggestion
You can use [the update route](/references/updates.md) to determine if a process has been committed to MeiliSearch or not.
```

> - status: `enqueued` => If MeiliSearch is killed, when restarted the task will still have enqueued as its status and will be processed eventually.

Reviewer: ```suggestion
- status: `enqueued` => Not yet begun. If MeiliSearch is killed and then restarted, the task will remain enqueued and be processed eventually.
```

> - status `processing` => If MeiliSearch is killed, when restarted, since it is atomic, there will be no consequences as none of the parts of the task has been committed to MeiliSearch, Meilisearch will start that task as if it was `enqueued`.

Reviewer: ```suggestion
- status `processing` => In progress. If MeiliSearch is killed, there will be no consequences, since no part of the task has been committed to MeiliSearch. After restarting, Meilisearch will treat the task as `enqueued`.
```

> - status `done` => If MeiliSearch is killed, this action is done and is permanently added to MeiliSearch. There is no data loss and your database/settings will be exactly the same as before you killed MeiliSearch.

Reviewer: ```suggestion
- status `done` => Completed. This action is done and is permanently added to your MeiliSearch instance. If you kill MeiliSearch, there will be no data loss; your database will remain exactly the same as before you killed MeiliSearch.
```

#### Example

> Imagine adding 100 documents in one batch to MeiliSearch. If you kill the process after 99 documents were correctly handled, when you restart MeiliSearch none of the 100 documents will be present in the dataset. Same goes if the 100th document raises an error.

Reviewer: ```suggestion
Imagine that you're adding 100 documents in one batch to MeiliSearch. If you kill the process after 99 documents have successfully been added, then none of the 100 documents will be present in the dataset when you restart MeiliSearch. The same is true if the 100th document raises an error. Either all documents are added, or none are.
```

> **Thus, killing MeiliSearch is safe!**

Reviewer: ```suggestion
Thus, **killing MeiliSearch is always safe!**
```
